From 33537c02f970f69713a03740bf2f707afc38960d Mon Sep 17 00:00:00 2001
From: Lu Baolu <baolu.lu@linux.intel.com>
Date: Wed, 8 Jun 2022 11:09:55 +0800
Subject: [PATCH 102/141] iommu: Export the IOMMU interface to the modular
 device driver

Otherwise the compiler will complains:

ERROR: modpost: "iommu_alloc_nested_domain" [drivers/iommu/iommufd/iommufd.ko] undefined!
ERROR: modpost: "iommu_domain_cache_inv" [drivers/iommu/iommufd/iommufd.ko] undefined!
ERROR: modpost: "iommu_attach_device_pasid" [drivers/iommu/iommufd/iommufd.ko] undefined!
ERROR: modpost: "iommu_detach_device_pasid" [drivers/iommu/iommufd/iommufd.ko] undefined!

Signed-off-by: Lu Baolu <baolu.lu@linux.intel.com>

4	0	drivers/iommu/iommu.c

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index 8c67fd46bb09..045f168bba7c 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -3165,6 +3165,7 @@ int iommu_attach_device_pasid(struct iommu_domain *domain,
 
 	return ret;
 }
+EXPORT_SYMBOL_GPL(iommu_attach_device_pasid);
 
 void iommu_detach_device_pasid(struct iommu_domain *domain,
 			       struct device *dev, ioasid_t pasid)
@@ -3178,6 +3179,7 @@ void iommu_detach_device_pasid(struct iommu_domain *domain,
 
 	iommu_group_put(group);
 }
+EXPORT_SYMBOL_GPL(iommu_detach_device_pasid);
 
 struct iommu_domain *iommu_get_domain_for_dev_pasid(struct device *dev,
 						    ioasid_t pasid)
@@ -3218,9 +3220,11 @@ iommu_alloc_nested_domain(struct bus_type *bus, struct iommu_domain *s2_domain,
 
 	return domain;
 }
+EXPORT_SYMBOL_GPL(iommu_alloc_nested_domain);
 
 void iommu_domain_cache_inv(struct iommu_domain *domain,
 			    struct iommu_cache_invalidate_info *inv_info)
 {
 	domain->ops->cache_invalidate(domain, inv_info);
 }
+EXPORT_SYMBOL_GPL(iommu_domain_cache_inv);
-- 
2.34.1

